internal/admission/admit_cluster_test.go

```diff
@@ -0,0 +1,336 @@
+
+	azcorearm "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+type expectedError struct {
+	message   string // Expected error message (partial match)
+	fieldPath string // Expected field path for the error
+}
+
+func verifyErrorsMatch(t *testing.T, expectedErrors []expectedError, errs field.ErrorList) {
+	if len(expectedErrors) != len(errs) {
+		t.Errorf("expected %d errors, got %d: %v", len(expectedErrors), len(errs), errs)
+		return
+	}
+
+	// Check that each expected error message and field path is found
+	for _, expectedErr := range expectedErrors {
+		if len(strings.TrimSpace(expectedErr.fieldPath)) == 0 {
+			t.Errorf("expected error with path %s to be non-empty", expectedErr.fieldPath)
+		}
+		if len(strings.TrimSpace(expectedErr.message)) == 0 {
+			t.Errorf("expected error with msg %s to be non-empty", expectedErr.message)
+		}
+
+		found := false
+		for _, err := range errs {
+			messageMatch := strings.Contains(err.Detail, expectedErr.message) || strings.Contains(err.Error(), expectedErr.message)
+			fieldMatch := strings.Contains(err.Field, expectedErr.fieldPath)
+			if messageMatch && fieldMatch {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("expected error containing message '%s' at field '%s' but not found in: %v", expectedErr.message, expectedErr.fieldPath, errs)
+		}
+	}
+
+	for _, err := range errs {
+		found := false
+		for _, expectedErr := range expectedErrors {
+			messageMatch := strings.Contains(err.Detail, expectedErr.message) || strings.Contains(err.Error(), expectedErr.message)
+			fieldMatch := strings.Contains(err.Field, expectedErr.fieldPath)
+			if messageMatch && fieldMatch {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("actual error '%v' but not found in expected", err)
+		}
+	}
+}
+
+// Helper function to create a valid cluster for testing
+func createValidCluster() *api.HCPOpenShiftCluster {
+	cluster := api.NewDefaultHCPOpenShiftCluster(api.Must(azcorearm.ParseResourceID("/subscriptions/0465bc32-c654-41b8-8d87-9815d7abe8f6/resourceGroups/some-resource-group/providers/Microsoft.RedHatOpenShift/hcpOpenShiftClusters/test-cluster")))
+
+	// Set required fields
+	cluster.Location = "eastus"
+	cluster.CustomerProperties.Version.ID = "4.15"
+	cluster.CustomerProperties.Version.ChannelGroup = "stable"
+	cluster.CustomerProperties.DNS.BaseDomainPrefix = "test-cluster"
+	cluster.CustomerProperties.Platform.SubnetID = "/subscriptions/0465bc32-c654-41b8-8d87-9815d7abe8f6/resourceGroups/some-resource-group/providers/Microsoft.Network/virtualNetworks/test-vnet/subnets/test-subnet"
+	cluster.CustomerProperties.Platform.NetworkSecurityGroupID = "/subscriptions/0465bc32-c654-41b8-8d87-9815d7abe8f6/resourceGroups/some-resource-group/providers/Microsoft.Network/networkSecurityGroups/test-nsg"
+	cluster.CustomerProperties.Platform.ManagedResourceGroup = "managed-rg"
+
+	return cluster
+}
```

Comment:
Not a blocker, but just FYI in `/internal/api/testhelpers.go` there's a function named `MinimumValidClusterTestCase` that does the same as this, except for the explicit version ID.

Could simplify if you want but I'll leave it up to you.